R/list-modify.R

```diff
@@ -42,32 +43,34 @@
 #' # a list of arguments:
 #' l <- list(new = 1, y = zap(), z = 5)
 #' str(list_update(x, !!!l))
-list_update <- function(.x, ...) {
+list_update <- function(.x, ..., .is_node = NULL) {
   check_list(.x)
-
   y <- dots_list(..., .named = NULL, .homonyms = "error")
-  list_recurse(.x, y, function(x, y) y, recurse = FALSE)
+
+  list_recurse(.x, y, function(x, y) y, recurse = FALSE, is_node = .is_node)
 }
 
 #' @export
 #' @rdname list_update
-list_modify <- function(.x, ...) {
+list_modify <- function(.x, ..., .is_node = NULL) {
   check_list(.x)
-
   y <- dots_list(..., .named = NULL, .homonyms = "error")
-  list_recurse(.x, y, function(x, y) y)
+
+  list_recurse(.x, y, function(x, y) y, is_node = .is_node)
 }
 
 #' @export
 #' @rdname list_update
-list_merge <- function(.x, ...) {
+list_merge <- function(.x, ..., .is_node = NULL) {
   check_list(.x)
-
   y <- dots_list(..., .named = NULL, .homonyms = "error")
-  list_recurse(.x, y, c)
+
+  list_recurse(.x, y, c, is_node = .is_node)
 }
 
-list_recurse <- function(x, y, base_f, recurse = TRUE, error_call = caller_env()) {
+list_recurse <- function(x, y, base_f, recurse = TRUE, error_call = caller_env(), is_node = NULL) {
+  is_node <- as_is_node(is_node, error_call, ".is_node")
```

Comment:
Maybe name the optional arguments here? `".is_node"` is the main one that took me a second to understand